### PR TITLE
Add Scale Object module

### PR DESCRIPTION
**Pull request by Kexanone**
**When merged this pull request will:**
- Add module for scaling objects (restricted to attached objects)

### CODE REVIEW

File: addons/modules/XEH_postInit.sqf [hunk 1]
```diff
@@ -99,3 +99,9 @@ if (isServer) then {
     _unit assignAsGunner _vehicle;
     [_unit] orderGetIn true;
 }] call CBA_fnc_addEventHandler;
+
+[QGVAR(setObjectScale), {
+    params ["_object", "_scale"];
+
+    _object setObjectScale _scale;
+}] call CBA_fnc_addEventHandler;
```
Comment (mharis001) [inline addons/modules/XEH_postInit.sqf:107]: For consistency, put single command events in common post init.

File: addons/common/stringtable.xml [hunk 2]
```diff
@@ -643,5 +643,10 @@
             <Korean>Zeus 카메라를 따라다니는 독수리의 생성 여부를 제어합니다</Korean>
             <Japanese>Zeus カメラ生成時に、イーグルがカメラを追随するかどうかを設定します。</Japanese>
         </Key>
+        <Key ID="STR_ZEN_Common_Scale">
+            <English>Scale</English>
+            <French>Redimensionnement</French>
+            <German>Skalierung</German>
+        </Key>
```
Comment (mharis001) [inline addons/common/stringtable.xml:650]: There is no proper organization to the stringtables but I'd say move this entry up to the other single-word ones to maintain setting names and descriptions at the bottom.

File: addons/modules/functions/fnc_moduleScaleObject.sqf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+if (isNull _object) exitWith {
+    [LSTRING(NoObjectSelected)] call EFUNC(common,showMessage);
+};
+
+if (isNull attachedTo _object) exitWith {
+    [LSTRING(NotAttachedTo)] call EFUNC(common,showMessage);
+};
+
+[LSTRING(ScaleObject), [
+    ["EDIT", [ELSTRING(common,Scale), LSTRING(ScaleObject_Tooltip)], "1"]
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleScaleObject.sqf:32]: `getObjectScale` for default value.

File: addons/modules/functions/fnc_moduleScaleObject.sqf [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+], {
+    params ["_values", "_object"];
+    _values params ["_scale"];
+
+    [QGVAR(setObjectScale), [_object,  parseNumber _scale], _object] call CBA_fnc_targetEvent;
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleScaleObject.sqf:37]: Scale is limited to 0.0001 to 65504. Should maybe `min`/`max` check that here.

File: addons/modules/stringtable.xml [hunk 5]
```diff
@@ -857,6 +857,16 @@
             <Japanese>オブジェクトの高さをこの値で変更します (負の値も可能).</Japanese>
             <Korean>이 값에 따라 물체의 고도를 변경합니다. (음수도 가능)</Korean>
         </Key>
+        <Key ID="STR_ZEN_Modules_ScaleObject">
+            <English>Scale Object</English>
+            <French>Redimensionner Objet</French>
+            <German>Objekt skalieren</German>
+        </Key>
+        <Key ID="STR_ZEN_Modules_ScaleObject_Tooltip">
+            <English>Scale Object by this factor.</English>
```
Comment (mharis001) [inline addons/modules/stringtable.xml:866]: Lowercase 'o' in object. Also, "**by** this factor" kind of implies that the new scale is affected by the current scale. Maybe change to "Scale the object by this factor relative to the original model."

File: addons/modules/functions/fnc_moduleScaleObject.sqf [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+if (isNull _object) exitWith {
+    [LSTRING(NoObjectSelected)] call EFUNC(common,showMessage);
+};
+
+if (isNull attachedTo _object) exitWith {
+    [LSTRING(NotAttachedTo)] call EFUNC(common,showMessage);
+};
+
+[LSTRING(ScaleObject), [
+    ["EDIT", [ELSTRING(common,Scale), LSTRING(ScaleObject_Tooltip)], str (getObjectScale _object)]
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleScaleObject.sqf:32]: No parentheses for unary commands.

File: addons/modules/functions/fnc_moduleScaleObject.sqf [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+], {
+    params ["_values", "_object"];
+    _values params ["_scale"];
+    _scale = parseNumber _scale;
+
+    if (_scale < OBJECT_SCALE_MIN || {_scale > OBJECT_SCALE_MAX}) exitWith {
+        [format [LLSTRING(ValueOutOfRange), OBJECT_SCALE_MIN, OBJECT_SCALE_MAX]] call EFUNC(common,showMessage);
+    };
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleScaleObject.sqf:40]: Blank line after `if` code block.

File: addons/modules/functions/fnc_moduleScaleObject.sqf [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+if (isNull attachedTo _object) exitWith {
+    [LSTRING(NotAttachedTo)] call EFUNC(common,showMessage);
+};
+
+[LSTRING(ScaleObject), [
+    ["EDIT", [ELSTRING(common,Scale), LSTRING(ScaleObject_Tooltip)], str (getObjectScale _object)]
+], {
+    params ["_values", "_object"];
+    _values params ["_scale"];
+    _scale = parseNumber _scale;
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleScaleObject.sqf:36]: Would prefer blank line between `params` stuff and rest of code.

**Review by mharis001 — COMMENTED** [review]
Looks good. Some minor cleanup suggestions.

**Review by neilzar — APPROVED** [review]

File: addons/modules/functions/fnc_moduleScaleObject.sqf [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    [LSTRING(NotAttachedTo)] call EFUNC(common,showMessage);
+};
+
+[LSTRING(ScaleObject), [
+    ["EDIT", [ELSTRING(common,Scale), LSTRING(ScaleObject_Tooltip)], str getObjectScale _object]
+], {
+    params ["_values", "_object"];
+
+    _values params ["_scale"];
+    _scale = parseNumber _scale;
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleScaleObject.sqf:37]:
```suggestion
    params ["_values", "_object"];
    _values params ["_scale"];

    _scale = parseNumber _scale;
```

**Review by CreepPork — APPROVED** [review]

File: addons/modules/functions/fnc_moduleScaleObject.sqf [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+if (isNull _object) exitWith {
+    [LSTRING(NoObjectSelected)] call EFUNC(common,showMessage);
+};
+
+if (isNull attachedTo _object) exitWith {
+    [LSTRING(NotAttachedTo)] call EFUNC(common,showMessage);
+};
+
+[LSTRING(ScaleObject), [
+    ["EDIT", [ELSTRING(common,Scale), LSTRING(ScaleObject_Tooltip)], str getObjectScale _object]
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleScaleObject.sqf:32]:
```suggestion
    ["EDIT", [ELSTRING(common,Scale), LSTRING(ScaleObject_Tooltip)], str getObjectScale _object, true]
```
To force default.

**Review by mharis001 — APPROVED** [review]

**Pull request merged**